src/visualizers/Turtle.ts

```diff
@@ -204,53 +307,240 @@ to prevent lag: this speed cannot exceed 1000 steps per frame.
+  will provide access to the value of the sequence being visualized for any
+  index. For example, the formula `A(n+1) - A(n)` or equivalently `A(n+1) - a`
+  would produce the so-called "first differences" of the sequence.
+
+  Note that as you change parameters in List mode, the formulas will be
+  updated in the background to match for you, but not vice versa: If you
+  switch to List mode, the visualization will change to reflect the last
+  List mode values you set. That's because most changes you might make
+  in Formula mode can't possibly be captured in the simpler List mode. Hence,
+  the Turtle visualizer simply doesn't try to keep the List values in
+  sync with Formula changes.
      **/
-    strokeWeight: {
-        default: 1,
-        type: ParamType.INTEGER,
-        displayName: 'Stroke width',
-        required: false,
-        validate: function (n: number, status: ValidationStatus) {
-            if (n <= 0) status.addError('must be positive')
+    ruleMode: {
+        default: RuleMode.List,
+        type: ParamType.ENUM,
+        from: RuleMode,
+        displayName: 'Rule mode',
+        required: true,
+        updateAction: function (newRule: string) {
+            const turtle = this instanceof Turtle ? this : null
+            if (turtle === null) return
+            const rule: RuleMode = parseInt(newRule)
+            if (rule === RuleMode.List) {
+                turtle.params.ruleMode.description = ''
+            } else {
+                turtle.params.ruleMode.description =
+                    'The formulas below may use the following symbols: '
+                    + '`a` - sequence entry, `n` - sequence index, '
+                    + '`s` - step number, `m` - minimum index, '
+                    + '`M` - Maximum index, '
+                    + '`A(...)` -- sequence entry at any index, '
+                    + '`b` - current bearing, `x`,`y` - current position, '
+                    + '`f` - frame number (triggers animation).'
+                // If there are any invalid rules, reset them to their
+                // default values:
+                const freshRules = new Set<string>()
+                let name: RuleParam = 'angles'
+                for (name in ruleParams) {
+                    if (turtle.statusOf[name].invalid()) {
+                        turtle.resetParam(name)
+                        freshRules.add(name)
+                    }
+                }
+                if (freshRules.size) turtle.refreshParams(freshRules)
+            }
         },
-        visibleDependency: 'pathLook',
-        visibleValue: true,
     },
     /** md
-- Background color: The color of the visualizer canvas.
-     **/
-    bgColor: {
-        default: '#6b1a1a',
-        type: ParamType.COLOR,
-        displayName: 'Background color',
+- Angle formula: an expression to compute the angle for a given step of the
+  turtle's path. Just as in list mode, the Angle is... parameter
```

Comment:
what's the ...?

Reply:
oh, I get it.  Maybe say the `Angle is....` parameter or something (my grammar parser just broke)

Reply:
How's this? I figured that any typography we applied to offset `Angle is...` should be applied uniformly to all parameter names. But then using all code-quotes was not very easy on the eyes, in part because in mkdocs the code bits come out red. So I did boldface... but it is now a lot of boldface. Would you prefer italics? Let me know if so and I will switch from bold to italics for the Turtle parameter names.

Reply:
I like it.